File: store/src/test/java/com/nytimes/android/external/store/KeyParserTest.java
package com.nytimes.android.external.store;

import com.nytimes.android.external.store.base.impl.Store;
import com.nytimes.android.external.store.base.impl.StoreBuilder;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.runners.MockitoJUnitRunner;

import rx.Observable;
import rx.observers.AssertableSubscriber;

@RunWith(MockitoJUnitRunner.class)
public class KeyParserTest {

    public static final String NETWORK = "Network";
    public static final int KEY = 5;
    private Store<String, Integer> store;

    @Before
    public void setUp() throws Exception {
        store = StoreBuilder.<Integer, String, String>parsedWithKey()
                .parser((integer, s) -> s + integer)
                .fetcher(integer -> Observable.just(NETWORK))
                .open();

    }

    @Test
    public void testStoreWithKeyParserFuncNoPersister() throws Exception {
        AssertableSubscriber<String> testObservable = store.get(KEY).test().awaitTerminalEvent();
        testObservable.assertNoErrors()
                .assertValues(NETWORK + KEY)
                .assertUnsubscribed();


    }
}

File: app/src/test/java/com/nytimes/android/sample/KeyParserTest.kt
package com.nytimes.android.sample

import com.nytimes.android.external.store3.base.Fetcher
import com.nytimes.android.external.store3.base.Parser
import com.nytimes.android.external.store3.base.impl.Store
import com.nytimes.android.external.store3.base.impl.StoreBuilder
import kotlinx.coroutines.runBlocking
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.runners.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class KeyParserTest {
  lateinit var store: Store<String, Int>

  @Before
  @Throws(Exception::class)
  fun setUp() {
    store = StoreBuilder.parsedWithKey<Int, String, String>()
        .parser(object : Parser<String, String> {
          override suspend fun apply(raw: String): String = raw + KEY
        })
        .fetcher(object : Fetcher<String, Int> {
          override suspend fun fetch(key: Int): String = NETWORK
        })
        .open()
  }

  @Test
  fun testStoreWithKeyParserFuncNoPersister() {
    runBlocking {
      assertThat(store.get(KEY)).isEqualTo(NETWORK + KEY)
    }
  }

  companion object {

    val NETWORK = "Network"
    val KEY = 5
  }
}
